### PR TITLE
Album art in player/album grid

### PR DESCRIPTION
**Pull request by Ruin0x11**

### CODE REVIEW

File: art.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+from mutagen.flac import FLAC
+from config import config
+import imghdr
+
+ART_DIR = config.get('Artwork', 'art_path')
+
+def index_art(song):
+    ext = splitext(song['path'])[1]
+
+
```
Comment (colegleason) [inline art.py]: remove extra whitespace

File: art.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    if isinstance(tags, FLAC) and tags.pictures:
+        data = tags.pictures[0].data
+    else:
+        for tag in tags:
+            if tag.startswith('APIC'):
+                data = tags[tag].data
+                break
+
+
```
Comment (colegleason) [inline art.py]: remove extra space

File: db.py [hunk 3]
```diff
@@ -112,6 +113,26 @@ class Vote(Base):
     user = Column(String(8), primary_key=True)
 
 
+class Playlist(Base):
```
Comment (kevinwang) [inline db.py]: I think you mixed in code from the feature-playlists branch.

File: art.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    image_type = imghdr.what(None, data)
+    ext = ''
+
+    if image_type == 'jpeg':
+        ext = 'jpg'
+    elif image_type == 'png':
+        ext = 'png'
+
+    filepath = u"{0}{1} - {2}.{3}".format('.' + ART_DIR, song['artist'], song['album'], ext)
```
Comment (colegleason) [inline art.py]: if the extension is unknown then this will append a period at the end. Can you make the period part of `ext`?

File: art.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    out = open(filepath, 'w')
+    out.write(data)
+    out.close()
+
+
+def get_art(artist, album):
+    if not album or not artist:
+        return None
+    ext = ['.jpg', '.png']
+    name = artist + " - " + album;
```
Comment (colegleason) [inline art.py]: remove semicolon

File: db.py [hunk 6]
```diff
@@ -45,7 +46,7 @@ def dictify(self):
             'length': self.length,
             'path': self.path,
             'tracknumber': self.tracknumber,
-            'play_count': self.play_count(),
```
Comment (colegleason) [inline db.py]: i don't think this was supposed to be removed from master, was it?

File: db.py (hunk 3 above)
Comment (colegleason) [inline db.py]: This should be in playlists, not this PR.

File: song.py [hunk 7]
```diff
@@ -127,7 +131,7 @@ def add_songs_in_dir(path, store_checksum=False):
                 except Exception:
                     song_obj['album'] = None
 
-                try: # Track number optional
+                try:  # Track number optional
```
Comment (colegleason) [inline song.py]: revert

File: song.py [hunk 8]
```diff
@@ -118,6 +120,8 @@ def add_songs_in_dir(path, store_checksum=False):
                 if store_checksum:
                     with open(filepath, 'rb') as song_file:
                         song_obj['checksum'] = md5_for_file(song_file)
+                else:
+                    song_obj['checksum'] = None
```
Comment (colegleason) [inline song.py]: is this part of art?

File: song.py [hunk 9]
```diff
@@ -62,14 +64,14 @@ def _prune_dir(path, prune_modified=False):
...
-def add_songs_in_dir(path, store_checksum=False):
+def add_songs_in_dir(path, store_checksum=True):
     """Update database to reflect the contents of the given directory.
 
     store_checksum: Whether or not to store an MD5 file checksum in order to
     update the song metadata in the database if the file is modified. Disabled
     by default because it makes scanning a lot slower.
     """
-    already_added = _prune_dir(path, prune_modified=store_checksum)
+    already_added = _prune_dir(unicode(path), prune_modified=store_checksum)
```
Comment (colegleason) [inline song.py]: is this part of album art?

File: song.py [hunk 10]
```diff
@@ -62,14 +64,14 @@ def _prune_dir(path, prune_modified=False):
     return remaining_paths
 
 
-def add_songs_in_dir(path, store_checksum=False):
+def add_songs_in_dir(path, store_checksum=True):
```
Comment (colegleason) [inline song.py]: is this part of album art?

File: static/beats.js [hunk 11]
```diff
@@ -211,7 +211,7 @@ function($scope, $http, $interval, $cookies)
     // Data
     //
 
-    var backendBase = '/beats/1104'
+    var backendBase = '/beats/1104';
```
Comment (colegleason) [inline static/beats.js]: remove semicolon

File: static/beats.js [hunk 12]
```diff
@@ -539,6 +539,42 @@ function($scope, $http, $interval, $cookies)
         });
     }
 
+    $scope.loadPlaylist = function(playlist)
```
Comment (colegleason) [inline static/beats.js]: part of playlists

File: static/beats.js [hunk 13]
```diff
@@ -571,6 +607,16 @@ function($scope, $http, $interval, $cookies)
 
     };
 
+    $scope.createPlaylist = function(name)
```
Comment (colegleason) [inline static/beats.js]: part of playlists

File: art.py [hunk 14]
```diff
@@ -0,0 +1,93 @@
+from os import path, listdir
+from os.path import isfile, splitext, dirname, join
```
Comment (colegleason) [inline art.py]: can this be consilidated?

Comment (colegleason) [inline art.py]: suggesting just `from os import path, listdir`

File: art.py [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+        else:
+            return None
+    except:
+        return None
+
+    data = ''
+
+    if isinstance(tags, FLAC) and tags.pictures:
+        data = tags.pictures[0].data
+    else:
```
Comment (colegleason) [inline art.py]: make this an `elif isinstance(tags, MP3)`

File: db.py [hunk 16]
```diff
@@ -1,9 +1,10 @@
 from sqlalchemy import create_engine
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy import Column, Integer, String, Unicode, Float, DateTime
-from sqlalchemy import ForeignKey
+from sqlalchemy import ForeignKey, UniqueConstraint
```
Comment (colegleason) [inline db.py]: revert

File: static/beats.js [hunk 17]
```diff
@@ -371,12 +371,6 @@ function($scope, $http, $interval, $cookies)
         $scope.errorMessage = 'An error has occurred.';
     };
 
-    $scope.addToPlayList = function(playlist, song)
```
Comment (colegleason) [inline static/beats.js]: revert

**Comment by colegleason** [comment]
LGTM

**Pull request merged**